### PR TITLE
removing locked_coin_til_epoch

### PR DESCRIPTION
**Pull request by healthydeve**
## Description 

Title

## Test Plan 

How did you test the new or updated feature?

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

**Comment by healthydeve** [comment]
Is this all dead code / is there any front-end stuff that still uses this field? @Jordan-Mysten

**Review by 666lcz — APPROVED** [review]

File: sdk/typescript/src/framework/framework.ts [hunk 1]
```diff
@@ -151,7 +151,6 @@ export type DelegationData = SuiMoveObject & {
       id: string;
       version: number;
     };
-    coin_locked_until_epoch: Option<SuiMoveObject>;
```
Comment (666lcz) [inline sdk/typescript/src/framework/framework.ts:154]: you should keep this and add a `// TODO: clean up after 0.34` for backward compatibility

Comment (healthydeve) [reply sdk/typescript/src/framework/framework.ts:154]: Done.

**Review by Jordan-Mysten — APPROVED** [review]
FE changes LGTM, we don't have anything depending on this

**Pull request merged**